tests/e2e/flowtest/mcg/test_mcg_namespace_disruptions.py

```diff
@@ -0,0 +1,306 @@
+                s3_obj=mcg_obj
+            )
+
+            logger.info(
+                f"Verifying integrity of objects "
+                f"after re-spinning: {self.labels_map[pod_to_respin]}"
+            )
+            compare_directory(awscli_pod, amount=3)
+
+        # S3 account
+        user = NoobaaAccount(mcg_obj, name=user_name, email=email, buckets=[ns_bucket])
+        logger.info(f"Noobaa account: {user.email_id} with S3 access created")
+
+        # Admin sets Public access policy(*)
+        bucket_policy_generated = gen_bucket_policy(
+            user_list=["*"],
+            actions_list=['GetObject'],
+            resources_list=[f'{ns_bucket}/{"*"}']
+        )
+        bucket_policy = json.dumps(bucket_policy_generated)
+
+        logger.info(f'Creating bucket policy on bucket: {ns_bucket} with wildcard (*) Principal')
+        put_policy = put_bucket_policy(mcg_obj, ns_bucket, bucket_policy)
+        logger.info(f'Put bucket policy response from Admin: {put_policy}')
+
+        logger.info(f'Getting bucket policy on bucket: {ns_bucket}')
+        get_policy = get_bucket_policy(mcg_obj, ns_bucket)
+        logger.info(f"Got bucket policy: {get_policy['Policy']}")
+
+        # MCG admin writes an object to bucket
+        logger.info(f'Writing object on bucket: {ns_bucket} by admin')
+        assert s3_put_object(mcg_obj, ns_bucket, object_key, data), "Failed: PutObject"
+
+        # Verifying whether Get operation is allowed to any S3 user
+        logger.info(
+            f'Get object action on namespace bucket: {ns_bucket} '
+            f'with user: {user.email_id}'
+        )
+        assert s3_get_object(user, ns_bucket, object_key), "Failed: GetObject"
+
+        # Upload files to NS target
+        logger.info(f"Uploading objects directly to ns resource target: {aws_res[0]}")
+        sync_object_directory(
+            awscli_pod,
+            src=MCG_NS_ORIGINAL_DIR,
+            target=f"s3://{aws_res[0]}",
+            signed_request_creds=aws_s3_creds
+        )
+
+        for pod_to_drain in self.labels_map:
+            pod_obj = pod.Pod(
+                **pod.get_pods_having_label(
+                    label=self.labels_map[pod_to_drain],
+                    namespace=defaults.ROOK_CLUSTER_NAMESPACE
+                )[0]
+            )
+
+            # Retrieve the node name on which the pod resides
+            node_name = pod_obj.get()['spec']['nodeName']
+
+            if awscli_node_name == node_name:
+                logger.info(
+                    f"Skipping node drain since aws cli pod node: "
+                    f"{awscli_node_name} is same as {pod_to_drain} "
+                    f"pod node: {node_name}"
+                )
+                continue
+
+            # Drain the node
+            drain_nodes([node_name])
+            wait_for_nodes_status([node_name], status=constants.NODE_READY_SCHEDULING_DISABLED)
+            schedule_nodes([node_name])
+
```

Comment:
post node drain, do you think we should also verify that all storage pods are in running state?

Reply:
ack